hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetQuery.java

```diff
@@ -0,0 +1,67 @@
+package org.eclipse.hawkbit.repository.jpa;
+
+import org.eclipse.hawkbit.repository.TargetFields;
+import org.eclipse.hawkbit.repository.TargetQuery;
+import org.eclipse.hawkbit.repository.TargetFilterQueryManagement;
+import org.eclipse.hawkbit.repository.exception.EntityNotFoundException;
+import org.eclipse.hawkbit.repository.jpa.model.JpaTarget;
+import org.eclipse.hawkbit.repository.jpa.rsql.RSQLUtility;
+import org.eclipse.hawkbit.repository.model.Target;
+import org.eclipse.hawkbit.repository.model.TargetFilterQuery;
+import org.eclipse.hawkbit.repository.rsql.VirtualPropertyReplacer;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Slice;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.orm.jpa.vendor.Database;
+import org.springframework.transaction.annotation.Transactional;
+
+@Transactional(readOnly = true)
+public class JpaTargetQuery implements TargetQuery {
```

Comment:
... existing `TargetFilterQueryManagement` could be renamed to `TargetFilterManagement` or maybe you'll find a even better name for it to avoid confusion